### DISCUSSION TITLE
Agent labels

### DISCUSSION THREAD

**Discussion by vladaurosh** [louislam/dockge#345] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

Hello.
Just started to use dockge, works really nice.

Is it possible to add agent labels? Currently agents are labeled as ip:port, is it possible to name the agents?

### ✔️ Solution

Agents should have a name, that should be visible in agent list and compose stacks as well, instead of ip:port

**Comment by sluggathor** [comment]
to add to this:
allow to set labels, tags, groups for remote agents

**Reply by jo-chemla** [reply]
Dockge Agent is a great way to deploy containers to other hosts from a single orchestrator dockge instance. 

Adding groups and tags/labels, plus the ability to deploy the same docker-compose stack(s) to a group (not only this host or other agents) composed of multiple agents at once would be really great. The way Portainer uses Edge Agents Groups is a bit convoluted (Static via selection of all agents, or Dynamic based on filtering via tags), but can be a good start for the inspiration - see the blog posts [part 1](https://www.portainer.io/blog/using-the-portainer-edge-agent-edge-groups-and-edge-stacks-part-1) / [part 2](https://www.portainer.io/blog/using-the-portainer-edge-agent-edge-groups-and-edge-stacks-part-1-0), [guide](https://downloads.portainer.io/edge_agent_guide.pdf) and [doc](https://docs.portainer.io/advanced/edge-agent). 

Not sure how such a feature can fit with @louislam current mindset of file-based compose files and "don't kidnap users with DB", but this surely would match the needs of users managing lots of agents.

**Comment by larswmh** [comment]
The original feature request is now covered by #414.

**Comment by ahmd-soliman** [comment]
+1 to this. I manage stacks across several Docker hosts (a NAS plus a few VMs) from a single git repo, and today agents let me pick a host per stack in the UI, but there's no way to declare which stacks belong on which host or to apply them automatically. A group or tag-based way to target agents, ideally defined in a plain file so it stays compatible with Dockge's file-based approach, would cover this. It would also pair well with the git sync in #471. Happy to test if that helps.
